frontend/src/components/WildcardSelect.vue

```diff
@@ -0,0 +1,221 @@
+      return {
+        wildcardVariablePart: {
+          required: `${this.wildcardTextFieldLabel} is required`
+        },
+        wildcardSelectedValue: {
+          required: `${this.wildcardSelectLabel} is required`
+        }
+      }
+    },
+    textFieldClass () {
+      return this.wildcardSelectedValue.startsWithWildcard ? 'textFieldStartClass' : 'textFieldEndClass'
+    },
+    wildcardSelectItemObjects () {
+      return map(this.wildcardSelectItems, item => {
+        let startsWithWildcard = false
+        let endsWithWildcard = false
+
+        let selectedValue = item
+        if (startsWith(item, '*')) {
+          startsWithWildcard = true
+        }
+        if (endsWith(item, '*')) {
+          endsWithWildcard = true
+        }
+        selectedValue = trim(item, '*')
+
+        return {
+          text: selectedValue,
+          value: selectedValue,
+          startsWithWildcard,
+          endsWithWildcard,
+          isWildcard: startsWithWildcard || endsWithWildcard
+        }
+      })
+    },
+    wildcardTextFieldLabel () {
+      return this.wildcardSelectedValue.startsWithWildcard ? 'Prefix' : 'Suffix'
+    },
+    wildcardSelectHint () {
+      if (!this.wildcardSelectedValue.isWildcard) {
+        return undefined
+      }
+      const label = this.wildcardSelectedValue.startsWithWildcard ? 'prefix' : 'suffix'
+      return `Selected wildcard value requires a ${label} which needs to be specified`
+    },
+    internalValue () {
+      if (this.wildcardSelectedValue.startsWithWildcard) {
+        return `${this.wildcardVariablePart}${this.wildcardSelectedValue.value}`
+      }
+      if (this.wildcardSelectedValue.endsWithWildcard) {
+        return `${this.wildcardSelectedValue.value}${this.wildcardVariablePart}`
+      }
+      return this.wildcardSelectedValue.value
+    }
+  },
+  methods: {
+    getErrorMessages (field) {
+      return getValidationErrors(this, field)
+    },
+    onInput () {
+      this.$v.wildcardVariablePart.$touch()
+      this.$v.wildcardSelectedValue.$touch()
+      if (this.valid !== !this.$v.$invalid) {
+        this.valid = !this.$v.$invalid
+        this.$emit('valid', this.valid)
+      }
+      this.$emit('input', this.internalValue)
+    },
+    setInternalValue (newValue) {
+      const matches = filter(this.wildcardSelectItemObjects, (item) => {
+        if (item.startsWithWildcard) {
+          return new RegExp('^.+' + item.value + '$').test(newValue)
+        }
+        if (item.endsWithWildcard) {
+          return new RegExp('^' + item.value + '.+$').test(newValue)
+        }
+        return new RegExp('^' + item.value + '$').test(newValue)
+      })
+      matches.sort(function (a, b) {
+        return b.value.length - a.value.length
+      })
+
+      const bestMatch = head(matches)
+      if (!bestMatch) {
+        return
+      }
+
+      this.wildcardSelectedValue = bestMatch
+      if (bestMatch.isWildcard) {
+        const value = trim(newValue, '*')
```

Comment:
do we still need to trim here? please double check

Reply:
Yes, still required, can explain if you want.

Reply:
is newValue the value from the selection? Because the value from the selection does not contain the wildcard, or does it?